### PR TITLE
Initial addition of asynchronous callbacks

### PR DESCRIPTION
**Pull request by embeddedt**

### CODE REVIEW

**Review by kisvegabor — CHANGES_REQUESTED** [review]
Thank you! Looks great in general. I have only a few simple remarks.

**Comment by kisvegabor** [comment]
Although it's a new feauture I think we merge it to `master` because it's required to conveniently use e.g. `LV_EVENT_DELETE`.
What do you think?

File: src/lv_misc/lv_async.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+lv_res_t lv_async_call(lv_async_cb_t async_xcb, void * user_data);
+
+/**
+ * Helper function for asynchronously deleting objects.
+ * 
+ * Useful for cases where you can't delete an object directly in an `LV_EVENT_DELETE` handler (i.e. parent).
+ * @param obj object to delete
+ * @see lv_async_call
+ */
+void lv_obj_del_async(struct _lv_obj_t *obj);
```
[the remark this answers is not in the document]
Comment (embeddedt) [reply src/lv_misc/lv_async.h:63]: @kisvegabor Yes, sorry for the mix-up. Will fix it later today.

**Comment by embeddedt** [comment]
@kisvegabor That's fine. It will have to be merged manually, though, or I will have to find some way of rebasing my commits.

I will make the remaining changes later today.

**Comment by kisvegabor** [comment]
> It will have to be merged manually, though, or I will have to find some way of rebasing my commits.

Why can't we simply merge it as usual?

**Comment by embeddedt** [comment]
I created the `async` branch based on `dev-6.1`.

**Comment by kisvegabor** [comment]
I see. I downloaded the patch from [here](https://github.com/littlevgl/lvgl/pull/1137.patch) and I could apply it to master (there were only some white space error.)

image

**Comment by embeddedt** [comment]
@kisvegabor In order for that to work, I have to change the base branch back to `dev-6.1` (which I just did). Otherwise it will still apply all of the previous commits.

**Comment by embeddedt** [comment]
I moved `lv_obj_del_async` and cherry-picked the changes to master.

**Comment by kisvegabor** [comment]
Great, thank you!
I've updated the docs: https://docs.littlevgl.com/en/html/overview/task.html#asynchronous-calls

**Comment by embeddedt** [comment]
> I've updated the docs

I like the description of the feature but the example seems to be a bit misleading. When you use `delay` in the `led_flash` function it makes it look like it's running on a parallel thread, when in reality it's going to block all execution.

**Comment by kisvegabor** [comment]
Ture. Do you have good and simple example idea?

**Comment by kisvegabor** [comment]
I've got an idea: we can add a `delay` parameter too to `lv_async_call` and return the created `lv_task_t`. 
 This way it can be used for a timeout too and can be deleted if the expected event happened. What do you think?

**Comment by embeddedt** [comment]
> Do you have good and simple example idea?

Deleting an object's parent is pretty straightforward.

> What do you think?

Well... at that point it would basically be a wrapper around `lv_task_create` that calls `lv_task_once`, right? I think for more advanced cases we should encourage people to use `lv_task_create` directly.

**Comment by kisvegabor** [comment]
> Deleting an object's parent is pretty straightforward.

I've updated the example: https://docs.littlevgl.com/en/html/overview/task.html#asynchronous-calls

> Well... at that point it would basically be a wrapper around lv_task_create that calls lv_task_once, right? I think for more advanced cases we should encourage people to use lv_task_create directly.

Alright. It's better to not change the API after the release anyway. :smile:

**Pull request closed without merging**